### PR TITLE
0.1x ci

### PR DESCRIPTION
**Pull request by technoweenie**
Backports our current CircleCI/CodeClimate config over from master. Hopefully we can then test and merge #1013.

### CODE REVIEW

**Comment by technoweenie** [comment]
Our test suite changes in master were never backported to this branch. Removing linting from the circle ci config was easy enough, but [now the tests don't run](https://circleci.com/gh/lostisland/faraday/1145?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link).

```
bundle exec rspec --require rspec_junit_formatter --format progress --format RspecJunitFormatter --out ~/test-results/rspec/results.xml

bundler: command not found: rspec
```
I think we have a couple options:

* Try to update this branch to run test/unit.
* Backport all of the test suite, circle ci, and rubocop changes to `0.1x`
* Release current master as v0.16.0 into the new stable branch `0.16.x`. This could essentially be our v1.0 RC branch.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    steps:
+      - checkout
+      - persist_to_workspace:
+          root: .
+          paths:
+            - .
+
+  ruby26:
+    docker:
+      - image: circleci/ruby:2.6
```
Comment (olleolleolle) [inline .circleci/config.yml]:
```suggestion
      - image: circleci/ruby:2.6-jessie
```

**Comment by technoweenie** [comment]
Closing in favor of just shipping v0.16.0 and a 0.16.x branch.

**Comment by BobbyMcWho** [comment]
@technoweenie do you wanna un-delete this branch? I can copy/paste these changes, but would rather just cherry-pick

**Comment by technoweenie** [comment]
This PR sets up Circle CI, which we've already transitioned from. So I don't think it's makes sense to bring this back.

**Comment by BobbyMcWho** [comment]
Ah, I wasn't aware that the project was moving away from circleci

**Pull request closed without merging**